backend/corpora/dataset_processing/process_cxg.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python3
+import logging
+import os
+import sys
+
+from backend.corpora.common.utils.s3_buckets import s3_client
+from backend.corpora.dataset_processing.process import (
+    check_env,
+    log_batch_environment,
+    update_db,
+    cancel_dataset,
+    notify_slack_failure,
+    process_cxg,
+    get_bucket_prefix,
+    LABELED_H5AD_FILENAME,
+)
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+from backend.corpora.common.corpora_orm import (
+    ProcessingStatus,
+)
+from backend.corpora.dataset_processing.exceptions import ProcessingCancelled, ProcessingFailed, ValidationFailed
+
+
+def download_from_s3(bucket_name: str, object_key: str, local_filename: str):
+    logger.info(f"Downloading file {local_filename} from bucket {bucket_name} with object key {object_key}")
+    s3_client.download_file(bucket_name, object_key, local_filename)
+
+
+def process(dataset_id: str, artifact_bucket: str, cellxgene_bucket: str):
+    """
+    1. Download the labeled dataset from the artifact bucket
+    1. Convert the labeled dataset to CXG
+    2. Upload the CXG to the cellxgene bucket
+    :param dataset_id:
+    :param artifact_bucket:
+    :param labeled_h5ad_filename:
+    :param cellxgene_bucket:
+    :return:
+    """
+
+    labeled_h5ad_filename = "local.h5ad"
+
+    # Download the labeled dataset from the artifact bucket
+    bucket_prefix = get_bucket_prefix(dataset_id)
+    object_key = f"{bucket_prefix}/{labeled_h5ad_filename}"
+    download_from_s3(artifact_bucket, object_key, labeled_h5ad_filename)
```

Comment:
Do any files persist in the compute environment after processing? In the past we were running out of space because these files were not getting cleaned up.

Reply:
The processing happens in a Docker environment so they won't persist.